python/lsst/ts/m2/csc.py

```diff
@@ -281,10 +279,14 @@ async def _event_loop(self):
 
         while self._run_loops:
 
+            message = (
+                self.model.queue_event.get_nowait()
+                if not self.model.queue_event.empty()
+                else await self.model.queue_event.get()
```

Comment:
`await self.model.queue_event.get()` is a blocking call. Therefore it will have the similar problem we discussed in the yesterday. However, if the `self.model.queue_event` is full, there should be something wrong in the controller or hardware to make it continuously report the event.

The main thing is that once it blocks in the edge case, how could we close the application actively? Or maybe we do not need to consider this because this is in CSC level?

Reply:
The `await` part will only be called if the queue is empty, which is the opposite of being full, hehehe.. 

In any case, `stop_loops` has a timeout for waiting the loops to stop, then they are cancelled, which will take care of the case when there's no new data arriving.